### PR TITLE
Build Mode Update!

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

Improves the experience of using build mode for building events and stuff.
You can now choose any from the eight dirs, not just cardinals and northwest.
Decals are now supported by build mode! This means that they can have any dir you want instead of always being south.
Areas are now filled in by default, you can still paint them manually with altclick.
Switches new tab width from 5 to 4, which means that the buttons for modes/dirs will be more naturally distributed, 4 per line when we have 8 dirs and 12 modes

## Why It's Good For The Game

Makes Build Mode not kill you instantly with cringe when you use it.

## Changelog
:cl: Fikou
admin: Build Mode Update!
qol: You can now choose any from the eight dirs, not just cardinals and northwest.
qol: Decals are now supported by build mode! This means that they can have any dir you want instead of always being south.
qol: Areas are now filled in by default, you can still paint them manually with altclick.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
Screenshots:
4 width between lines now
image
Dirs working for decals
image
8 dir support
image
Area fill-in
image
image

**Comment by Zytolg** [comment]
Not the build mode we needed, but the build mode we deserved. Excellent.

**Review by Twaticus — APPROVED** [review]

File: code/modules/buildmode/buildmode.dm [hunk 1]
```diff
@@ -72,7 +72,7 @@
 	buttons += new /atom/movable/screen/buildmode/quit(src)
 	// build the lists of switching buttons
 	build_options_grid(subtypesof(/datum/buildmode_mode), modeswitch_buttons, /atom/movable/screen/buildmode/modeswitch)
-	build_options_grid(list(SOUTH,EAST,WEST,NORTH,NORTHWEST), dirswitch_buttons, /atom/movable/screen/buildmode/dirswitch)
+	build_options_grid(list(SOUTH,NORTH,EAST,WEST,SOUTHEAST,SOUTHWEST,NORTHEAST,NORTHWEST), dirswitch_buttons, /atom/movable/screen/buildmode/dirswitch)
```
Comment (LemonInTheDark) [inline code/modules/buildmode/buildmode.dm:75]: shouldn't this be GLOB.alldirs?

File: code/modules/buildmode/submodes/advanced.dm [hunk 2]
```diff
@@ -47,6 +47,10 @@
 			log_admin("Build Mode: [key_name(c)] modified [T] in [AREACOORD(object)] to [objholder]")
 			T = T.ChangeTurf(objholder)
 			T.setDir(BM.build_dir)
+		else if(ispath(objholder, /obj/effect/turf_decal))
+			var/turf/T = get_turf(object)
+			T.AddElement(/datum/element/decal, initial(objholder.icon), initial(objholder.icon_state), BM.build_dir, FALSE, initial(objholder.color), null, null, initial(objholder.alpha))
```
Comment (LemonInTheDark) [inline code/modules/buildmode/submodes/advanced.dm:52]: can't you achieve this by spawning in the decal itself, and letting it do the work?

File: code/modules/buildmode/submodes/fill.dm [hunk 3]
```diff
@@ -64,6 +64,8 @@
 				if(ispath(objholder,/turf))
 					T = T.ChangeTurf(objholder)
 					T.setDir(BM.build_dir)
+				else if(ispath(objholder, /obj/effect/turf_decal))
+					T.AddElement(/datum/element/decal, initial(objholder.icon), initial(objholder.icon_state), BM.build_dir, FALSE, initial(objholder.color), null, null, initial(objholder.alpha))
```
Comment (LemonInTheDark) [inline code/modules/buildmode/submodes/fill.dm:68]: ditto here

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/buildmode/buildmode.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/buildmode/buildmode.dm:75]:
```suggestion
	build_options_grid(GLOB.alldirs, dirswitch_buttons, /atom/movable/screen/buildmode/dirswitch)
```

**Pull request merged**